### DISCUSSION TITLE
특정 프로젝트 페이지 할일의 상태 추가 등록에 대하여 궁금한 점이 있습니다.

### DISCUSSION THREAD

**Discussion by Seok93** [GU-99/grow-up-fe#23] [질문과 답변]
### 할일 등록 기능
할일을 등록하면 이를 캘린더 양식 or 칸반 보드 양식으로 볼 수 있도록 UI를 제공하는 것으로 이해하고 있습니다. 이때 할일이라는 데이터가 기본적으로 `To Do`, `In Progress`,  `Done` 등의 상태를 가지고 있고, 추가적으로 상태 정보를 추가할 수 있죠. 그런데 이러한 상태값 추가 등록 기능이 현재 칸반 보드에만 있습니다. 

 할일이라는 데이터는 공유되고, 데이터 등록시 할일의 상태를 선택하도록 되어 있으니, 필요시에 캘린더 양식에서 등록할 때에도 상태값을 추가할 수 있도록 만드는게 좋지 않을까요?

p.s 추가적으로 할일 등록 버튼도 별도로 빼는건 어떨까요?

---
#### 🖼️ Kanban 양식 화면
![칸반](https://github.com/GU-99/grow-up-fe/assets/32609010/97ab9b54-c751-4233-ad44-5ef120e4aeb7)

---
#### 🖼️ Calendar 양식 화면
![캘린더](https://github.com/GU-99/grow-up-fe/assets/32609010/38b63783-ed8a-4948-b8f5-f29046ff247a)

**Comment by kimpearl3599** [comment]
할일 등록버튼 있으면 좋을 것 같습니다 :)

**Comment by Yoonyesol** [comment]
그러네요 할일 추가는 양쪽에서 모두 가능하게 수정하는 것이 좋겠네요! 그런데 할 일 버튼을 별도로 뺀다는 건 버튼이 공통으로 쓰이게 되면 분리하는 게 좋겠다는 말씀이신가요??😳

**Reply by Seok93** [reply]
> 버튼이 공통으로 쓰이게 되면 분리하는 게 좋겠다는 말씀이신가요??😳

제가 이 말을 이해를 못해서... 죄송합니다.  버튼을 공용 컴포넌트로 만든다는 의미로 말씀하신건가요? 
저는 `+New State`버튼처럼 옆에 할일 추가 버튼을 만드는게 어떠냐는 의미였어요. 지금은 캘린더의 특정 날짜를 누르거나 칸반 보드에서 `+`버튼을 눌러서 등록하는 형태로 되어 있는것 같아서요.

**Reply by Yoonyesol** [reply]
넵 맞습니다. 저도 이해를 제대로 하지 못해서 그런 의미로 ps 남겨주신 건가 하고 질문드렸습니다😂
우석님 말씀대로 New State 버튼 추가로 만들면 좋을 것 같아요!

**Reply by Seok93** [reply]
그렇군요! 유사한 디자인과 기능도 비슷해서 공용 컴포넌트로 빼는 것도 괜찮을 것 같아요!  만들 때 고려해보겠습니다😄

**Comment by Seok93** [comment]
음... 사실 이 토론의 주요 논지는 `상태 추가 등록(상태명/색상)`입니다💦 
물론 `할일 추가 등록`도 p.s로 적어두긴 했지만요😅 제가 전달을 잘못했나봐요 ㅋㅋㅋㅋ
`상태 추가 등록` 버튼이랑 `할일 추가 등록`  버튼 둘 다 캘린더와 칸반에서 가능하도록 만드는게 좋을 것 같아요!

**Reply by Yoonyesol** [reply]
네 칸반이랑 캘린더 두 개는 각자 뷰만 달라지는 거니까 상태 추가나 할일 추가 버튼은 동일하게 만드는 게 좋을 것 같습니다!
